nomos-services/tracing/src/lib.rs

```diff
@@ -250,16 +250,24 @@ where
 
             let level_filter = {
                 #[cfg(feature = "profiling")]
-                if let ConsoleLayer::Console(console_config) = &config.console
-                    && let Some(console_layer) = console::create_console_layer(console_config)
                 {
-                    layers.push(console_layer);
-                    LevelFilter::TRACE
-                } else {
-                    LevelFilter::from(config.level)
+                    if let ConsoleLayer::Console(console_config) = &config.console {
+                        if let Some(console_layer) = console::create_console_layer(console_config) {
+                            layers.push(console_layer);
+                            LevelFilter::TRACE
+                        } else {
+                            LevelFilter::from(config.level)
+                        }
+                    } else {
+                        LevelFilter::from(config.level)
+                    }
                 }
+
+                // Fallback for Windows or when profiling feature is disabled:
                 #[cfg(not(feature = "profiling"))]
-                LevelFilter::from(config.level)
+                {
+                    LevelFilter::from(config.level)
+                }
             };
```

Comment:
I think the extra scopes don't provide any value. Also, splitting the conditions on the `if` is unnecessary and a bit confusing.